wolfcrypt/src/sha512.c

```diff
@@ -506,8 +528,13 @@ static INLINE void AddLength(Sha512* sha512, word32 len)
 
 int wc_Sha512Update(Sha512* sha512, const byte* data, word32 len)
 {
+    byte* local;
```

Comment:
This change is a good example of "declaration then assignment". You changed the line:

```byte* local = (byte*)sha512->buffer;``` and did this:


```
byte* local;

<error check>

local = (byte*)sha512->buffer;
```